src/Options/OptionsDetails.hpp

```diff
@@ -92,7 +92,7 @@ using options_in_group =
 // information.
 template <typename T>
 struct yaml_type {
-  static std::string value() { return pretty_type::short_name<T>(); }
+  static std::string value() { return pretty_type::name<T>(); }
```

Comment:
In the commit message you say that sometimes it would be better to use `short_name` still, "for example in tags". Why so?

Reply:
If you have 
```
struct MyTag {
  static std::string name() { return pretty_type::name<MyTag>(); }
};
```

`pretty_type::name` will end up in a recursive call to the tag's `name()` function. Because of compile time stuff, I'm not sure if this is actually recursive or if it ends, but when getting the name of a struct from within the `name()` member itself I felt it better to use `short_name` instead to avoid this all together.

Reply:
In these cases, shouldn't we just delete the `name` function and make sure the consuming code uses `pretty_type::name`?

Reply:
Yes I suppose I can change `DataBox/TagName.hpp` to use `pretty_type::name` now and remove the name member from tags that just have `short_name<Tag>` implemented. Still doesn't prevent somebody in the future from making a new tag that uses `pretty_type::name` inside the `name()` member though...

Reply:
DataBox tags currently should only have a `name` function defined if `name` returns something other than `pretty_type::short_name<Tag>`   There are `TestHelpers::db::test_TAGTYPE_tag` functions that check this if the tags are properly tested.  So if you change the behavior, please update the test helpers...

Reply:
and if you find untested tags, please test them...

Reply:
My suggestion is to warn not to use `pretty_type::name` inside a `name` function if this can lead to recursion. That's fairly obvious, but why not add a warning. We don't have to deal with changing any DataBox / tags things in this PR, but if you want to make a decision then I'm not stopping you :)

Reply:
Sorry, what do you mean by "add a warning"? Do you mean in doxygen or somehow in the code? If the latter I'm not sure how to implement it, so providing some detail would probably be useful.

Reply:
@nilsdeppe I meant in docs